src/lyd_mods.c

```diff
@@ -1069,6 +1069,41 @@ sr_lydmods_ctx_load_modules(const struct lyd_node *sr_mods, struct ly_ctx *ly_ct
     return NULL;
 }
 
+/**
+ * @brief Learn whether any module in a set augments/deviates a specific module.
+ *
+ * @param[in] old_base_mod Module that can be augmented/deviated.
+ * @param[in] old_mod_set Set with modules possibly augmenting/deviating @p old_base_mod.
+ * @return Whether any module in @p old_mod_set was augmenting/deviating @p old_base_mod.
+ */
+static int
+sr_contains_dev_aug_module(const struct lys_module *old_base_mod, struct ly_set *old_mod_set)
+{
+    const struct lys_module *old_mod;
+    uint32_t i;
+    LY_ARRAY_COUNT_TYPE u;
+
+    for (i = 0; i < old_mod_set->count; ++i) {
+        old_mod = old_mod_set->objs[i];
+
+        /* deviates */
+        LY_ARRAY_FOR(old_base_mod->deviated_by, u) {
+            if (old_base_mod->deviated_by[u]->name == old_mod->name) {
```

Comment:
Um, sorry for not being more clear. I mean, this should work, too, but what I meant was comparing the module pointers directly, no need to compare names, that is just redundant and confusing.

Reply:
ok